pgmpy/factors/Factor.py

```diff
@@ -191,39 +191,26 @@ def marginalize(self, variables, inplace=True):
             factor = self
         else:
             factor = Factor(self.scope(), self.cardinality, self.values)
-
+        marginalize_index = np.array(np.where(np.in1d(factor.scope(),variables)))
+        assign = np.array(factor.cardinality)
+        assign[marginalize_index] = -1
+        sum_variable = np.zeros(len(factor.cardinality))
+        sum_variable[marginalize_index] = -1
```

Comment:
Since you are using `assign[assing != -1]` in the for loop, it returns a
new object. So changing the value of `assign` inside the loop shouldn't
affect it.

On Wed, Feb 25, 2015 at 4:16 PM, palashahuja <notifications@github.com>
wrote:

> In pgmpy/factors/Factor.py
> <https://github.com/pgmpy/pgmpy/pull/282#discussion_r25331173>:
>
> > @@ -191,39 +191,26 @@ def marginalize(self, variables, inplace=True):
> >              factor = self
> >          else:
> >              factor = Factor(self.scope(), self.cardinality, self.values)
> > -
> > +        marginalize_index = np.array(np.where(np.in1d(factor.scope(),variables)))
> > +        assign = np.array(factor.cardinality)
> > +        assign[marginalize_index] = -1
> > +        sum_variable = np.zeros(len(factor.cardinality))
> > +        sum_variable[marginalize_index] = -1
>
> Can we really do that ? Won't that change the value of assign inside the
> loop ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pgmpy/pgmpy/pull/282/files#r25331173>.
>
> --
> You received this message because you are subscribed to the Google Groups
> "pgmpy" group.
> To unsubscribe from this group and stop receiving emails from it, send an
> email to pgmpy+unsubscribe@googlegroups.com.
> To post to this group, send email to pgmpy@googlegroups.com.
> To view this discussion on the web visit
> https://groups.google.com/d/msgid/pgmpy/pgmpy/pgmpy/pull/282/r25331173%40github.com
> <https://groups.google.com/d/msgid/pgmpy/pgmpy/pgmpy/pull/282/r25331173%40github.com?utm_medium=email&utm_source=footer>
> .
>
> For more options, visit https://groups.google.com/d/optout.
>